spec.bs

```diff
@@ -1509,18 +1521,38 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
     to the result of running [=is debugging only in cooldown or lockout=] with |buyer|.
   1. [=map/For each=] |signalsUrl| → |perSignalsUrlGenerator| of |perSlotSizeQueryParam|:
     1. [=map/For each=] |slotSizeQueryParam| → |perSlotSizeQueryParam| of |perBuyerGenerator|:
+      1. Let |allTrustedBiddingSignals| be an [=ordered map=].
       1. Let |keys| be a new [=ordered set=].
       1. Let |igNames| be a new [=ordered set=].
+      1. Let |lengthLimit| be the maximum value of {{long}} interger.
       1. Let |fetchSignalStartTime| be |settings|'s [=environment settings object/current monotonic time=].
       1. [=map/For each=] joiningOrigin → |groups| of |perSignalsUrlGenerator|:
         1. [=list/For each=] |ig| of |groups|:
-          1. [=set/Append=] |ig|'s [=interest group/trusted bidding signals keys=] to |keys|.
-          1. [=set/Append=] |ig|'s [=interest group/name=] to |igNames|.
-      1. Let |biddingSignalsUrl| be the result of [=building trusted bidding signals url=] with
-        |signalsUrl|, |keys|, |igNames|, |buyerExperimentGroupId|, |topLevelOrigin|, and
-        |slotSizeQueryParam|.
-      1. Let « |allTrustedBiddingSignals|, |dataVersion| » be the result of [=fetching trusted signals=]
-        with |biddingSignalsUrl| and true.
+          1. Set |keys| to |putativeKeys|.
+          1. Set |igNames| to |putativeIgNames|.
+          1. Set |lengthLimit| to |putativeLengthLimit|, if |lengthLimit| &le; |ig|'s

```

Comment:
this step does not seem correctly matching the [implementation](https://crsrc.org/c/content/services/auction_worklet/trusted_signals_request_manager.cc;drc=8333e7c470463683707ca13e57a838cd7e608405;l=233). Can you double check if the current steps here match our implementation?

Reply:
I did use the set to in an opposite way. Here is the updated code, which will set the smaller number to `putativeLengthLimit` between the `lengthLimit` from last loop and the current `maxTrustedBiddingSignalsURLLength` of the current IG:

```
Set |putativeLengthLimit| to |lengthLimit|, if |lengthLimit| &le; |ig|'s
            [=interest group/max trusted bidding signals url length=]. Otherwise, set |putativeLengthLimit| to
            |ig|'s [=interest group/max trusted bidding signals url length=].
```